src/components/implementation/no_interface/scheddev/sl.h

```diff
@@ -174,10 +193,163 @@ sl_cs_exit(void)
 	if (!ps_cas(&sl__globals()->lock.u.v, cached.v, 0))    goto retry;
 }
 
+/*
+ * if tid == 0, just block the current thread; otherwise, create a
+ * dependency from this thread on the target tid (i.e. when the
+ * scheduler chooses to run this thread, we will run the dependency
+ * instead (note that "dependency" is transitive).
+ */
+void sl_thd_block(thdid_t tid);
+/*
+ * if abs_timeout == 0, block forever = sl_thd_block()
+ *
+ * @returns: 0 if the thread is woken up by external events before timeout.
+ *	     +ve - number of cycles elapsed from abs_timeout before the thread
+ *		   was woken up by Timeout module.
+ */
+cycles_t sl_thd_block_timeout(thdid_t tid, cycles_t abs_timeout);
+/*
+ * blocks for on periodic wakeup based on task period.
+ *
+ * @returns: 0 if the thread is woken up by external events before timeout.
+ *           +ve - number of periods elapsed. (1 if it wokeup exactly at timeout = next period)
+ */
+unsigned int sl_thd_block_periodic(thdid_t tid);
+int  sl_thd_block_no_cs(struct sl_thd *t, sl_thd_state block_type);
+/* wakeup a thread that has (or soon will) block */
+void sl_thd_wakeup(thdid_t tid);
+int  sl_thd_wakeup_no_cs(struct sl_thd *t);
+void sl_thd_yield(thdid_t tid);
```

Comment:
Hmm, block/wakeup/yield are actions on threads and we've had them grouped until now. 
But yeah, now that we have so many versions of each, we might as well have space between each action. Will do!